tests/cost_report/infrastructure/test_infrastructure_profile_costs_calculator.py

```diff
@@ -0,0 +1,89 @@
+import math
+
+from koswat.cost_report.infrastructure.infrastructure_location_costs import (
+    InfrastructureLocationCosts,
+)
+from koswat.cost_report.infrastructure.infrastructure_profile_costs_calculator import (
+    InfrastructureProfileCostsCalculator,
+)
+from koswat.dike.surroundings.surroundings_infrastructure import (
+    SurroundingsInfrastructure,
+)
+from tests.conftest import PointSurroundingsTestCase
+
+
+class TestInfrastructureProfileCostsCalculator:
+    def test_initialize(self):
+        # 1. Define test data.
+        _calculator = InfrastructureProfileCostsCalculator()
+
+        # 2. Verify expectations.
+        assert isinstance(_calculator, InfrastructureProfileCostsCalculator)
+        assert not _calculator.infrastructure
+        assert math.isnan(_calculator.surtax_costs)
+        assert math.isnan(_calculator.zone_a_costs)
+        assert math.isnan(_calculator.zone_b_costs)
+
+    def test_given_infrastructure_fixture_calculates_costs(
+        self,
+        surroundings_infrastructure_fixture: tuple[
+            SurroundingsInfrastructure, PointSurroundingsTestCase
+        ],
+    ):
+        """
+        This test validates that the matrix costs (e.g.: {0: a, 5: b, 10: c})
+        will be used so that based on the width of the `zone_a` we round up its value
+        until a key match is found.
+        Example given:
+            - if `zone_a = 4` then range is `[0, 4]` and:
+                - matrix keys `0` an `5`, both included,
+                - `zone_a_width = a + b`.
+            - if `zone_b = 1` then range `[4, 5]` and:
+                - matrix keys `10` as `0` and `5` were claimed by `zone_a`.
+                - `zone_b_width = c`.
+        """
+        # 1. Define test data.
+        # Set costs to 1 for easy comparisons.
+        _surtax_costs = 1.0
```

Comment:
Setting to 1 might cover problems with calculations. The difference between dividing by 1, multiplying by 1 and not taking it into account will not be uncovered.

Reply:
That's not correct, we are not using surtax to calculate anything. Is just a simple property that gets carried on for later calculations at the `CostReportProtocol` level, so that when calculating the total cost of "something" we can decide whether to take this `surtax` value or not. At least is how I understood it, otherwise we need to check with PO.